File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int n) 
    {
        List<int> collatz = new List<int>();
        collatz.Add(n);
        while (collatz[collatz.Count-1] != 1) {

            if (collatz[collatz.Count-1]%2 == 0) {
                collatz.Add(collatz[collatz.Count-1]/2); 
            }
            else if (collatz[collatz.Count-1]%2 == 1) {
                 collatz.Add(3 * collatz[collatz.Count-1] + 1); 
            }
        }
        
        int[] answer = collatz.ToArray();  
        return answer; 
    }
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) 
{
    vector<int> answer;
    answer.push_back(n);
    
    while(answer.back() != 1)
    {
        int curr = answer.back();
        if(curr%2 == 0)
        {
            answer.push_back(curr/2);
        }
        else
        {
            answer.push_back(curr * 3 + 1);
        }
    }
    return answer;
}